stdlib/Test/src/Test.jl

```diff
@@ -712,24 +736,30 @@ function do_test_throws(result::ExecutionResult, orig_expr, extype)
                 else
                     isa(exc, extype)
                 end
-        else
-            if extype isa LoadError && !(exc isa LoadError) && typeof(extype.error) == typeof(exc)
-                extype = extype.error # deprecated
-            end
-            if isa(exc, typeof(extype))
-                success = true
-                for fld in 1:nfields(extype)
-                    if !isequal(getfield(extype, fld), getfield(exc, fld))
-                        success = false
-                        break
-                    end
+        elseif isa(exc, typeof(extype))
+            success = true
+            for fld in 1:nfields(extype)
+                if !isequal(getfield(extype, fld), getfield(exc, fld))
+                    success = false
+                    break
                 end
             end
+        elseif isa(extype, Exception)
+        else
```

Comment:
```suggestion
        elseif !isa(extype, Exception)
```
(nit: looks odd otherwise, like something was missing)